pkg/client/unversioned/clientcmd/client_config.go

```diff
@@ -273,66 +293,79 @@ func (config *DirectClientConfig) ConfirmUsable() error {
 	return newErrConfigurationInvalid(validationErrors)
 }
 
-func (config *DirectClientConfig) getContextName() string {
+func (config *DirectClientConfig) getContextName() (string, bool) {
+	requireCurrentContext := true
+	if config.overrides.IgnoreMissingContext {
+		requireCurrentContext = false
+	}
+
 	if len(config.overrides.CurrentContext) != 0 {
-		return config.overrides.CurrentContext
+		return config.overrides.CurrentContext, requireCurrentContext
 	}
 	if len(config.contextName) != 0 {
-		return config.contextName
+		return config.contextName, false
 	}
 
-	return config.config.CurrentContext
+	return config.config.CurrentContext, false
 }
 
-func (config *DirectClientConfig) getAuthInfoName() string {
+func (config *DirectClientConfig) getAuthInfoName() (string, bool) {
 	if len(config.overrides.Context.AuthInfo) != 0 {
-		return config.overrides.Context.AuthInfo
+		return config.overrides.Context.AuthInfo, true
 	}
-	return config.getContext().AuthInfo
+	context, _ := config.getContext()
+	return context.AuthInfo, false
 }
 
-func (config *DirectClientConfig) getClusterName() string {
+func (config *DirectClientConfig) getClusterName() (string, bool) {
 	if len(config.overrides.Context.Cluster) != 0 {
-		return config.overrides.Context.Cluster
+		return config.overrides.Context.Cluster, true
 	}
-	return config.getContext().Cluster
+	context, _ := config.getContext()
+	return context.Cluster, false
 }
 
-func (config *DirectClientConfig) getContext() clientcmdapi.Context {
+func (config *DirectClientConfig) getContext() (clientcmdapi.Context, error) {
 	contexts := config.config.Contexts
-	contextName := config.getContextName()
+	contextName, required := config.getContextName()
 
 	var mergedContext clientcmdapi.Context
 	if configContext, exists := contexts[contextName]; exists {
 		mergo.Merge(&mergedContext, configContext)
+	} else if required {
```

Comment:
@liggitt 
> I'm still not seeing why the test needs to send an overridden "current context" at all, if it's overriding the actual context values

I went ahead and removed the `CurentContext` override from the test. I kept the new `IgnoreMissingContext` [flag as a way to tell client_config when to 'require' a context](https://github.com/kubernetes/kubernetes/pull/29236/commits/6adf162344be6d38f55fede3e016b33724918fa1#diff-093eaa515b227f1070ec0fda3f8e0137R297) that has been overridden (the kubectl integration test otherwise still fails without it). PTAL